pkg/cmd/step/e2e/step_e2e_gc.go

```diff
@@ -156,6 +252,37 @@ func (o *StepE2EGCOptions) ShouldDeleteOlderThanDuration(cluster *gke.Cluster) b
 	return false
 }
 
+// ShouldDeleteDueToNewerRunEKS returns true if a cluster with a higher build number exists
+func (o *StepE2EGCOptions) ShouldDeleteDueToNewerRunEKS(cluster *cluster.Cluster, clusters []*cluster.Cluster) bool {
+	if branchLabel, ok := cluster.Labels["branch"]; ok {
+		if strings.Contains(branchLabel, "pr-") {
+			currentBuildNumber, err := o.GetBuildNumberFromClusterEKS(cluster)
+			if err == nil {
+				if clusterType, ok := cluster.Labels["cluster"]; ok {
+					for _, existingCluster := range clusters {
+						// Check for same PR & Cluster type
+						if existingClusterType, ok := existingCluster.Labels["cluster"]; ok {
+							if strings.Contains(existingCluster.Name, branchLabel) && existingClusterType == clusterType {
+								existingBuildNumber, err := o.GetBuildNumberFromClusterEKS(existingCluster)
+								if err == nil {
+									// Delete the older build
+									if currentBuildNumber < existingBuildNumber {
+										if _, ok := cluster.Labels["keep-me"]; !ok {
```

Comment:
Any chance we can decompose this a bit? Maybe by breaking out some functions which also describe a bit better what's going on here.

Reply:
This is pretty much a work in progress, I took the code for GKE and created equivalent code for EKS for now.

With the cluster.interface abstraction, it should be easy to keep the same code for every provider, but testing this for GKE and EKS at the same time would be too much. I created an issue to go back and do it right.

Reply:
but who knows what this is even doing? IMO this code is impossible to maintain. Could we not refactor this code and let the refactored code form the base for other providers and potentially let the refactor flow back to GKE?